hs_collection_resource/views.py

```diff
@@ -117,13 +117,14 @@ def update_collection(request, shortkey, *args, **kwargs):
                     = authorize(request, res_id_add,
                                 needed_permission=ACTION_TO_AUTHORIZE.VIEW_METADATA)
 
-                # the resources being added should be 'Shareable'
+                # the resources being added should be 'Shareable' or 'Public'
                 # or is owned by current user
                 is_shareable = res_to_add.raccess.shareable
+                is_public = res_to_add.raccess.public
                 is_owner = res_to_add.raccess.owners.filter(pk=user.pk).exists()
-                if not is_shareable and not is_owner:
-                        raise Exception('Only resource owner can add a non-shareable '
-                                        'resource to a collection ')
+                if not is_shareable and not is_public and not is_owner:
+                    raise Exception('Only resource owner can add a non-shareable or private '
+                                    'resource to a collection ')
```

Comment:
@pkdash @alvacouch @sblack-usu Should we allow discoverable resources to be added into collection resources as well? I believe the code here only allows public and published resources, NOT the discoverable resources, to be added into collection resources.

Reply:
@hyi I am not sure on this.

Reply:
@pkdash We discussed this on the coordination call and the team has decided that we should allow discoverable resources to be added in the collection in the same way as public resources.

Reply:
@hyi Sounds good. Will implement that change soon.

Reply:
@hyi I have now made code changes to allow discoverable to be part of a collection resource.